File: Java/File.java
import java.util.*;



public class File {

    private ArrayList<Integer> file = new ArrayList<Integer>();




    public static int alea(int min , int max){

        int nombreAleatoire = min + (int)(Math.random() * ((max - min) + 1));

        return (nombreAleatoire);




    }

    public synchronized String ajoute(){
        int nb_alea = alea(0,100);
        String resultat="";

        if(file.size() ==20){
            try{
            wait();
            }catch(Exception e){}
        }

        else {

            file.add(nb_alea);
            notifyAll();
            resultat="Ajout de l'entier "+nb_alea+" ...";
            return resultat;
            

        }

        return resultat;
        

        
}

    public synchronized String supprime(){
        String resultat="";

        if (file.isEmpty()){
            try{
            wait();
            }catch(Exception e){}
        }
        
        else {
            file.remove(0);
            notifyAll();
            resultat = "Retrait de "+file.get(0)+"effectue ! ";
            return resultat;


        }
        return resultat;

    }
    
   

    public String fileToString(){

        String fileString="";

        fileString+= "File : <";
        for (int i=0 ; i<file.size()-1 ; i++){

            fileString += (file.get(i)).toString() + "," ;


        }

        fileString+= (file.get(file.size()-1)).toString() + "<" ;
        
        
        return fileString;
        
    }



}

File: Python/file.py
from random import randint
from threading import Condition

class File ():

    def __init__(self):
        self.file= []
        self.cond = Condition()
    

    def ajoute(self):
        with self.cond:
            nb_alea = randint(0,100)
            resultat=""
            if (len(self.file) ==20 ):
                self.cond.wait()
            else:
                self.file.append(nb_alea)
                self.cond.notify_all()
                resultat="Ajout de l'entier "+nb_alea+" ..."
                return resultat
            return resultat
    
    def supprime(self):
        with self.cond :
            resultat=""
            if (len(self.file) ==0):
                self.cond.wait()
            else :
                self.file.pop()
                self.cond.notify_all()
                resultat = "Retrait de "+self.file[0]+" effectue ! "
                return resultat
            return resultat
    
    def fileToString(self):
        file_string=""

        for e in (self.file[0:-1]):
            file_string+=  str(e) + ','
        
        file_string+= str(self.file[-1])

    
        


